sunpycube/cube/NDCube.py

```diff
@@ -54,37 +65,50 @@ def plot(self, *args, **kwargs):
             If None is specified for an axis then the array indices will be used
             for that axis.
         """
-        i = ImageAnimatorWCS(self.data, wcs=self.axes_wcs, *args, **kwargs)
+        i = ImageAnimatorWCS(self.data, wcs=self.wcs, *args, **kwargs)
         return i
 
     def __getitem__(self, item):
         if item is None or (isinstance(item, tuple) and None in item):
             raise IndexError("None indices not supported")
-        return NDCube_utils.getitem_ND(self, item)
+        return self._getitem_NDCube(item)
+
+    def _getitem_NDCube(self, item):
+        data = self.data[item]
+        # wcs = self.wcs.wcs_slicer[item]
+        wcs = None
+        mask = None
+        if self.mask is not None:
+            mask = self.mask[item]
+        if data.ndim is 2:
+            result = Cube2D(data, wcs=wcs, mask=mask, uncertainty=self.uncertainty,
+                            meta=self.meta, unit=self.unit, copy=False)
+        elif data.ndim is 1:
+            result = Cube1D(data, wcs=wcs, mask=mask, uncertainty=self.uncertainty,
+                            meta=self.meta, unit=self.unit, copy=False)
+        else:
+            result = self._new_instance(data, wcs=wcs, mask=mask, uncertainty=self.uncertainty,
+                                        meta=self.meta, unit=self.unit, copy=False)
+        return result
```

Comment:
Does the ```copy=False``` kwarg mean that the ```result``` of ```_getitem_NDCube``` points to the original instance?  So if I do NDCube[0,:,:] = np.nan``` or something, I actually change the data in the original NDCube not a new instance?  That's the behaviour we want.